### ISSUE TITLE
Access to fetch at xxx from origin 'app://obsidian.md' has been blocked by CORS policy

### ISSUE THREAD

**Issue by hilsonp** [vrtmrz/obsidian-livesync#174]
Hello,

This is strange...
On Laptop2, from my office (then going through a corporate proxy), synchronization will not happen and my log is full of:
```
Access to fetch at 'https://obsidian-livesync.mydomain.be/xxx/h%3A%2B-795epx?revs=true&latest=true&open_revs=%5B%221-cd3c1cc4fcd6f9f7ade8630cd813ed1f%22%5D' from origin 'app://obsidian.md' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
plugin:obsidian-livesync:17192          GET https://obsidian-livesync.mydomain.be/xxx/h%3A%2B-795epx?revs=true&latest=true&open_revs=%5B%221-cd3c1cc4fcd6f9f7ade8630cd813ed1f%22%5D net::ERR_FAILED 307
```

If I take a Chrome browser and open https://obsidian-livesync.mydomain.be/, it connects fine and shows:
`{"couchdb":"Welcome","version":"3.3.1","git_sha":"1fd50b82a","uuid":"c09e56976e2a7d6f088426473e93ea7b","features":["access-ready","partitioned","pluggable-storage-engines","reshard","scheduler"],"vendor":{"name":"The Apache Software Foundation"}}`

I don't know when this happened (or if it ever worked from my office) as I was concentrated on my Laptop1 and my iPhone (with Obsidian crashing during sync).

I tried the "Use new adapter" but the problem remain.

Your comment https://github.com/vrtmrz/obsidian-livesync/issues/161#issuecomment-1406260795 let me think that this may be due to small packets (due to the fact that I had to change the Custom Chunk Size to 10 to avoid the iPhone crash.

I'm a bit lost on this CORS issue and I don't want to upset the proxy/firewall people. They could easily ban my domain and I would be stuck for many other self-hosted services 😥
So before I continue fidling with my settings on my (corporate) Laptop2, I kindly ask you if you have any clue/idea of things to try first.

**Comment by vrtmrz** [comment]
Thank you for your information!

While looking at that, I found it says `net::ERR_FAILED 307`.  It probably means `307 Temporary redirect`. 
The address mentioned in the log is accessing the document (probably, a chunk), but this API would not respond in 307 (For your information: [API Reference](https://docs.couchdb.org/en/stable/api/document/common.html#db-doc)).

However, you had been able to be reached by the database in the browser. Following the circumstance, it is possibly a response of the gateway; like detecting the high-frequent accessing or hitting the rate limit. 

CouchDB usually responds along with the CORS header, even if the request has an error and answers so. I have seen several errors like this, sent by such interceptors. Some gateway does not process the headers.

I supposed that balancing the request rate would be a bit harder than we thought, We should prepare the cap or delay...

> They could easily ban my domain and I would be stuck for many other self-hosted services 😥

Absolutely! It is also my circumstance. I can excuse myself, but not anytime be accepted.

In other options, we can disable `Read chunks online` to reduce request frequency, but the transfer amount is a bit larger. 
When this option is disabled, chunks will be distributed to the client while synchronisation. Otherwise, fetch from the remote database on demand. This fetching caused the 307 error.

**Comment by Gurgarath** [comment]
I will have to second this issue as I am also facing it. I thought it was because of cloudflare, my VPN or custom DNS service but disabling them had the exact same effect on the error and it remained. 
It works perfectly fine in iPhone and Windows, I use the exact same configuration on all platforms.

The error occurs on Arch Linux, everything up-to-date.

If needed:
Electron v25.1.1
Chromium v114.0.5735.90
Node v18.15.0v8 
v11.4.183.17-electron.0

**Comment by vrtmrz** [comment]
Thank you so much for the latest information! 
I have understood things, and I hope to close the issue once. However, feel free to open this issue again if there are any new topics.  (Of course, we can make a new issue!)

**Comment by chauser10** [comment]
> I will have to second this issue as I am also facing it. I thought it was because of cloudflare, my VPN or custom DNS service but disabling them had the exact same effect on the error and it remained. It works perfectly fine in iPhone and Windows, I use the exact same configuration on all platforms.
> 
> The error occurs on Arch Linux, everything up-to-date. I currently use Cloudflare for the certificate, CORS is well enabled for everything (*) on my CouchDB and it works on iOS, Windows but not Arch.
> 
> If needed: Electron v25.1.1 Chromium v114.0.5735.90 Node v18.15.0v8 v11.4.183.17-electron.0
> 
> Edit: As I previously thought, it was cloudflare. My Linux User Agent was blacklisted and somehow adding rules to bypass checks didn't work. Issue is solved and was from my side.

@Gurgarath  - You mentioned it was cloudflare. I am using a tunnel to access my couchDB instance. what did you change to fix this error?

**Comment by Gurgarath** [comment]
> @Gurgarath - You mentioned it was cloudflare. I am using a tunnel to access my couchDB instance. what did you change to fix this error?

I do not have neither the set-up nor the fix currently, but what I did was simply to check cloudflare logs to see that there was an attempted connection that was blocked. I then managed to make the User Agent bypass the safety rules into the WAF settings, I basically allowed this exact user agent to proceed without challenge (this is bad practice depending on your set-up and so on).
In your case, I would check the logs to see if there is a botched attempt to connect, also check your iptables / pf rules or any firewall if you have them on your set-up.
